AudioManagerUtilities/src/CAmSocketHandler.cpp

```diff
@@ -608,20 +669,22 @@ am_Error_e CAmSocketHandler::removeTimer(const sh_timerHandle_t handle)
     if (it == mListTimer.end())
         return (E_NON_EXISTENT);
 
-    close(it->fd);
+    mListRemovedTimers.push_back(*it);
     mListTimer.erase(it);
     return removeFDPoll(handle);
 #else
     stopTimer(handle);
     std::list<sh_timer_s>::iterator it(mListTimer.begin());
-    for (; it != mListTimer.end(); ++it)
+    while (it != mListTimer.end())
     {
         if (it->handle == handle)
         {
-            it = mListTimer.erase(it);
+            it = mListTimer.erase(it);            
             mSetTimerKeys.pollHandles.erase(handle);
             return (E_OK);
         }
+        else
```

Comment:
If you stick with the new implementation the else is superfluous.
Just increment the iterator.